### PR TITLE
AnonCreds Credentials using the W3C Standard - drop mapping from presentation

### PR DESCRIPTION
**Pull request by Artemkaaas**

### CODE REVIEW

File: src/data_types/credential.rs [hunk 1]
```diff
@@ -73,14 +77,71 @@ impl Validatable for Credential {
...
     pub referent: String,
-    pub attrs: ShortCredentialValues,
+    pub attrs: RawCredentialValues,
     pub schema_id: SchemaId,
     pub cred_def_id: CredentialDefinitionId,
     pub rev_reg_id: Option<RevocationRegistryDefinitionId>,
     pub cred_rev_id: Option<String>,
 }
 
-pub type ShortCredentialValues = HashMap<String, String>;
```
Comment (gvelez17) [inline src/data_types/credential.rs:83]: curious why is this necessary?

**Review by TimoGlastra — APPROVED** [review]
LGTM, but would be good if @berendsliedrecht could also take a look

File: src/ffi/credential.rs [hunk 2]
```diff
@@ -228,3 +196,446 @@ pub extern "C" fn anoncreds_credential_get_attribute(
...
+///
+/// # Params
+/// cred:       object handle pointing to W3C AnonCreds credential
+/// context:    context to add into credential
+/// cred_p:     reference that will contain update credential
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_w3c_credential_add_context(
```
[the remark this answers is not in the document]
Comment (Artemkaaas) [reply src/ffi/credential.rs]: I do not have strong vision and opinion regarding these methods. It may be clear during the integration of anoncreds-rs into framework. In general, the goal of adding these methods was supporting the case when an additional NonAnonCreds identity proof needs to be added.
The flow would be issue w3c credential -> do non-anoncreds signature on the application level -> append signature to the object -> serialize and store. 
If you consider these methods as redundant, we can drop them.

[the remark this answers is not in the document]
Comment (TimoGlastra) [reply src/ffi/credential.rs]: Please see my comments on #273, there's some relation to this comment.

@swcurran might be good for the WG call later today

File: src/ffi/cred_req.rs [hunk 3]
```diff
@@ -58,6 +58,44 @@ pub extern "C" fn anoncreds_create_credential_request(
     })
 }
 
+/// Create Credential Request according to the AnonCreds specification
+/// Note that Credential Request still will be legacy styled (the same as result of anoncreds_create_credential_request)
+///
+/// # Params
+/// entropy:                entropy string to use for request creation
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply src/ffi/cred_req.rs]: I think how the credential offer e.g. looks is protocol specific, so e.g. something we define in the Aries RFCs. Or if we want to integrate it into OpenID4VC there would need to be a protocol specific mapping for that.

I don't think there can be a w3c offer, as the w3c has no protocols for exchange, it just defines the data model

[the remark this answers is not in the document]
Comment (Artemkaaas) [reply src/ffi/cred_req.rs]: I removed these methods (duplicating offer/request creation) and non-anoncreds proof helper in the scope of PR:  https://github.com/hyperledger/anoncreds-rs/pull/273

**Comment by Artemkaaas** [comment]
Closed in favour of: https://github.com/hyperledger/anoncreds-rs/pull/291
